### ISSUE TITLE
Success Criterion 3.3.2 - Labels or Instructions - Level A

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#45]
WCAG2ICT guidance: https://www.w3.org/TR/wcag2ict-22/#labels-or-instructions

Share your thoughts for applying to mobile apps as a comment below.

**Comment by Keanem6** [comment]
I don't see any variation on this one.

**Comment by JJdeGroot** [comment]
We might want to clarify "label" in mobile context. But the definition as is seems pretty clear.

label:
> [text](https://www.w3.org/TR/WCAG22/#dfn-text) or other component with a [text alternative](https://www.w3.org/TR/WCAG22/#dfn-text-alternative) that is presented to a user to identify a component within Web [content](https://www.w3.org/TR/WCAG22/#dfn-content)
>
> Note 1
> A label is presented to all users whereas the [name](https://www.w3.org/TR/WCAG22/#dfn-name) may be hidden and only exposed by assistive technology. In many (but not all) cases the name and the label are the same.
>
> Note 2
> The term label is not limited to the label element in HTML.

The note 2 already makes it clear that it's not limited to the `<label>` HTML element.

And note 1 makes it clear that the label should be visible (whether text, or component with alt-text)

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>23 October 2024</summary>

[Source](https://www.w3.org/2024/10/23-matf-minutes.html#c9b0)

~~~text
We could probably add a note that labels should not disappear when input focus is given to the labelled element (i.e. hint)

<JJ> Hey all, sorry for being late - construction workers started making noise at the office so headed home, but missed the meeting notification because I was away from my laptop. Thanks for taking over Joe_Humbert!

Could we just add a relates link?

Joe_Humbert it seems that mobile developers do miss this

julianmka said there is a mention of this somewhere else

Jamie if we scope it to mobile apps, agrees

<JJ> Yes, note can help to clarify best-practices in mobile (app) context

Just add a contact in iPhone - the name field is an example of this error

julianmka there isn't a mention in 3.2.2 in the understanding docs

Joe_Humbert do we need to modify the definition of label?

Android would be "text" element, "UILabel" in IOS

(or "TextView" if you're a dinosaur Android dev)

<Jamie> +1 as is

<JJ> +1 as is

<quintinb> +1 as is

<julianmka> +1

<GleidsonRamos> +1 as is

<Karla> +1

ACTION: Suggest to accept 3.3.2 for the wider group

~~~
</details>

ACTION: Suggest to accept 3.3.2 for the wider group

**Comment by julianmka** [comment]
Based on previous task force conversation, this SC can be applied to native mobile apps and mobile web with minimal or no deviation from WCAG2ICT.

Proposal
In MATF's first draft of guidance, this SC's section will read as:

>This applies directly as written, and as described in [Intent from Understanding Success Criterion 3.3.2](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions#intent).

Please indicate your agreement with a thumbs up emoji 👍, or if you disagree, use the thumbs down emoji 👎 and elaborate in comments.

**Comment by JJdeGroot** [comment]
Hi @dotjay , thanks for contributing! As discussed in our meeting just now, we will be closing issues that reached consensus for our draft note. I'm going to close this issue now, but in a future stage we will get back to the points you mentioned.

Closing this issue because the draft language has been accepted.

**Comment by detlevhfischer** [comment]
I think there are similar cases where requirements are exempt when elements are employed without change that are defined by the user agent (or OS). See target size exeptions for UA controls. The argument in favour of exempting custom controls like this would be that users know these elements from other apps and the OS itself, and authors can therefore expect them to be understood. It also avoids forcing authors to craft some custom stuff because of issues with OS or UA level controls, stuff that may be harder to maintain (and that may also deviate from user expectations). 

So even where issues can be found in OS or UA-supplied date pickers like `input type="date"` in the web context (as is the case with some browsers) I think it would be bad to fail them and implicity force authors wanting or needing to conform to build a complex custom datepicker.

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>13 November 2024</summary>

[Source](https://www.w3.org/2024/11/13-matf-minutes.html#603b)

~~~text
Illai Are we refering to the entire component or the spinner alone for this SC?

JJ Google and Apple need to make more efforts to make the native components accessibile

Joe_Humbert the example looks like just a pop-up and that should be fine. However, the spinner is overlaid on calendar, which could be a problem for some users.

<Illai> +1 JJ

<Karla> +1 JJ

JJ this needs to be updated by Apple

<JJ> Do we agree to keep the original proposal text for 3.3.2 by Julian?

<Illai> +1

<Joe_Humbert> +1

<Aash> +1

<Karla> +1

<Tim> +1

Aash did not understand the 4 orientations concepts?

@jj

Joe_Humbert No more work needed to support all 4 orientations. It is same rendering in terms of pixels

JJ it is a best practice to support all 4 orientations.

Illai should we propose this in conjunction with target size SC?

<JJ> Should we add a note to 1.3.4 about the best-practice of supporting all 4 (or more) orientations of a device?

JJ we can add 4 orientation in notes, along with the target size SC, basic and advanced.

<Illai> +1

<Aash> +1

<Karla> +1

<GleidsonRamos> +1

<Tim> +1
~~~
</details>

Conclusion: we considered the concerns of @dotjay and @detlevhfischer ; however at this moment we don't think an exception is needed for the case that Jon presented. The time picker has a visible label, the time spinner itself is a pop-up. There might be more situations where we don't want to force authors to create their own components. If we identify situations where that's needed, we can add an exception in a future version of our guidance.

For now, we are going to accept the text proposed by Julian for our first draft.
